enhancements/olm/runtime-bundle-configuration.md

```diff
@@ -0,0 +1,858 @@
+- Define registry+v2 bundle configuration handling
+- Occlude secret information
+- Support for default configurations outside what can be set in a JSON-schema defined field default values
+- Support OLMv0 SubscriptionConfig-type behavior, enabling broader operator configuration compatibility, though this will come in the future
+- Support the MultiNamespace install mode: this mode can severally impact api server resource utilization and cause cluster instability. Even in OLMv0 it is documented as unwise to use and that it will likely be removed in the future (OLMv1 being that future).
+
+## Proposal
+
+### 1. Expand ClusterExtension API to Include Bundle Configuration
+
+The ClusterExtension API surface will be expanded to ingest opaque bundle configuration in the form of arbitrary JSON
+objects. This moves the concern of bundle configuration from OLM to the bundle author, paving the way for bundles
+to provide their own configuration schema. This opaque configuration is utilized by the bundle rendering layer independently
+of the underlying bundle format. For example, it can serve as the values.json for Helm bundles.
+
+The configuration will be supplied inline directly in the ClusterExtension API. The changes to the API will be designed
+in such a way that other sources of configuration (e.g. ConfigMaps, or Secrets) can be subsequently added if needed
+in the future.
+
+### 2. Update Bundle Manifest Provisioning Layer
+
+The bundle manifest provisioning layer is currently responsible for providing the bundle manifests that should be applied
+to the cluster. This interface will be expanded to accept the arbitrary configuration coming from the API, validate it, and
+account for it when generating the final set of manifests. The given configuration must conform to the configuration schema
+provided by the bundle. If the bundle cannot be configured, or the configuration does not strictly comply with the
+schema, an error is raised that prevents installation/upgrade/update from proceeding until the user can address the
+issue, e.g.:
+- User provides configuration but the bundle does not support configuration
+- User does not provide required configuration fields
+- User provides configuration keys that are not existent
+- User provides configuration but bundle does not provide a configuration schema - note that 
+while for registry+v1 bundles this is deemed a good level of restriction to ship the configuration feature with initially, for future bundle formats this restriction may be loosed (eg OLMv1 could allow users to provide configuration for "Helm bundles" without any configuration schema). Users will likely be required to express their intent to configure such bundles explicity and have OLM install anyway, possibly via a ClusterExtension API field (eg `dangerouslyAcceptBundlesWithoutConfigSchema`). Such considerations however are for future considerations. 
```

Comment:
Do we have an idea where in the API we'd put knobs like this? Do we need to make sure there is space in the `spec.config` struct since this is a config-specific knob? Or do we envision putting this kind of thing directly in `.spec`?